src/events.py

```diff
@@ -53,6 +53,25 @@ def get_event_steps(self, start_time, n_intervals, interval):
 
         return steps
 
+    def get_external_load_perfect_forseight(self, start_time, interval):
```

Comment:
What is the purpose of this function? Please add comments. Might not even be needed (see part of flex_window where this is called). Also, "foresight" is spelled wrong.